recipes-ni/ni-base-system-image-tests/files/fs_permissions_known.py

```diff
@@ -0,0 +1,153 @@
+import argparse
+import csv
+import datetime
+import hashlib
+import mmap
+import os
+import re
+import subprocess
+import sys
+
+class Logger:
+    prefix_logs = []
+    logs = []
+    def log(self, log):
+        self.logs.append(log)
+
+    def prefix_log(self, log):
+        self.prefix_logs.append(log)
+
+    def report(self):
+        for log in self.prefix_logs:
+            print(log)
+        for log in self.logs:
+            print(log)
+
+def run_cmd(cmd):
+    output = subprocess.check_output(cmd)
+    return output.decode('utf-8')
+
+# Columns: Path, Mode, User, Group, Link Target
+fs_manifest_format = '%p\t%M\t%u\t%g\t%l'
+def get_fs_manifest():
+    search_dirs = ['/bin', '/boot', '/etc', '/lib', '/lib64', '/sbin', '/usr', '/var']
+    omit_dirs = ['/etc/natinst/niskyline/Data/Assets/Cache', '/lib/modules', '/var/cache', '/var/run', '/var/tmp', '/var/volatile']
+
+    omit_expr = []
+    for d in omit_dirs:
+        omit_expr += ['-path', d, '-o']
+    omit_expr.pop()
+
+    fs_manifest = run_cmd(
+        ['find']
+        + search_dirs
+        + ['-printf', fs_manifest_format + '\n']
+        + ['('] + omit_expr + [')', '-prune']
+    )
+    entries = {}
+    for entry in fs_manifest.splitlines():
+        entry_data = entry.split('\t')
+        entries[entry_data[0]] = {
+            'mode': entry_data[1],
+            'user': entry_data[2],
+            'group': entry_data[3],
+            'link': entry_data[4]
+        }
+    return entries
+
+def log_version_info(logger, label, codename, full_version, date, db_id):
+    logger.log(f'INFO: {label} = {codename} {full_version} from {date} with _id {db_id}')
+
+def check_known_manifests(manifest, logger):
+    logger.log('###### Checking expected file permissions against actual for known folders')
+    failed = False
+    with open('known-permissions.csv') as known_perm_file:
+        permissions = csv.reader(known_perm_file, delimiter = ',')
```

Comment:
I'd suggest using a [`DictReader`](https://docs.python.org/3/library/csv.html#csv.DictReader) to make it more readable later, rather than using magic numbers as indices into the rows.

```suggestion
        permissions = csv.DictReader(known_perm_file, delimiter = ',', fieldnames = ['path', 'mode', 'user', 'group', 'link'])
```